### PR TITLE
OKTA-436460: Accept activation token

### PR DESCRIPTION
**Pull request by arvindkrishnakumar-okta**
Note this PR will be merged into `rc_2.0.0` branch and moved into the main branch when the APIs officially support this feature.

### CODE REVIEW

File: api/src/main/java/com/okta/idx/sdk/api/client/IDXClient.java [hunk 1]
```diff
@@ -27,12 +27,14 @@
...
 
+import java.util.Optional;
+
 /**
  * Client to interact with the IDX backend APIs.
  */
 public interface IDXClient {
 
-    IDXClientContext interact() throws ProcessingException;
+    IDXClientContext interact(Optional<String> activationToken) throws ProcessingException;
```
Comment (JayNewstrom) [inline api/src/main/java/com/okta/idx/sdk/api/client/IDXClient.java:37]: This is a binary API change. Should we do a new major version?

File: api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java [hunk 2]
```diff
@@ -672,16 +683,32 @@ public boolean isSkipAuthenticatorPresent(ProceedContext proceedContext) {
...
+            return AuthenticationTransaction.create(client, Optional.empty()).asAuthenticationResponse();
+        } catch (ProcessingException e) {
+            return handleProcessingException(e);
+        } catch (IllegalArgumentException e) {
+            return handleIllegalArgumentException(e);
+        }
+    }
+
+    /** Begin transaction with activation token **/
+    public AuthenticationResponse begin(String activationToken) {
```
Comment (JayNewstrom) [inline api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java:698]: I didn't see any new tests for this. Can we add one?

File: api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java [hunk 3]
```diff
@@ -560,6 +561,16 @@ public IDXClientContext getClientContext() throws ProcessingException {
         }
     }
 
+    /** Cancel the transaction with supplied state handle **/
+    public void cancel(String stateHandle) {
```
Comment (JayNewstrom) [inline api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java:565]: I didn't see any new tests for this. Can we add one?

File: api/src/main/java/com/okta/idx/sdk/api/client/IDXClient.java (hunk 1 above)
Comment (arvindkrishnakumar-okta) [reply api/src/main/java/com/okta/idx/sdk/api/client/IDXClient.java:37]: yep, it'd be a major rev. I'll update the rc branch name.

File: api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java (hunk 2 above)
Comment (arvindkrishnakumar-okta) [reply api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java:697]: https://github.com/okta/okta-idx-java/blob/okta-436460-add-activation-token/api/src/test/groovy/com/okta/idx/sdk/api/client/IDXAuthenticationWrapperTest.groovy#L363

Comment (JayNewstrom) [reply api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java:697]: There aren't any new assertions there. If you removed the code that sends the query param, all the tests would still pass.

File: api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java (hunk 3 above)
Comment (arvindkrishnakumar-okta) [reply api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java:564]: This API is out of scope in this PR, will take it out.

File: api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java (hunk 2 above)
Comment (arvindkrishnakumar-okta) [reply api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java:687]: Updated.

File: api/src/test/groovy/com/okta/idx/sdk/api/client/IDXAuthenticationWrapperTest.groovy [hunk 4]
```diff
@@ -348,6 +348,25 @@ class IDXAuthenticationWrapperTest {
...
+    @Test
+    void beginTransactionWithActivationTokenTest() {
+
+        def requestExecutor = mock(RequestExecutor)
+        def idxClient = new BaseIDXClient(getClientConfiguration(), requestExecutor)
+        def idxAuthenticationWrapper = new IDXAuthenticationWrapper()
+        //replace idxClient with mock idxClient
+        setInternalState(idxAuthenticationWrapper, "client", idxClient)
+
+        setMockResponseWithBodyParamMatcher(requestExecutor, "interact", "activation_token", "interact-response", 200, MediaType.APPLICATION_JSON)
```
Comment (JayNewstrom) [inline api/src/test/groovy/com/okta/idx/sdk/api/client/IDXAuthenticationWrapperTest.groovy:360]: Can we also test that it doesn't exist in the original use case?

Comment (arvindkrishnakumar-okta) [reply api/src/test/groovy/com/okta/idx/sdk/api/client/IDXAuthenticationWrapperTest.groovy]: done

File: api/src/main/java/com/okta/idx/sdk/api/client/WrapperUtil.java [hunk 5]
```diff
@@ -33,6 +33,7 @@
     static AuthenticationResponse handleIllegalArgumentException(IllegalArgumentException e) {
         AuthenticationResponse authenticationResponse = new AuthenticationResponse();
         logger.error("Exception occurred", e);
+        authenticationResponse.addError(e.getMessage());
```
Comment (JayNewstrom) [inline api/src/main/java/com/okta/idx/sdk/api/client/WrapperUtil.java:36]: Should this stay? Seems like it might have only been used for debugging.

**Review by JayNewstrom — APPROVED** [review]

File: api/src/main/java/com/okta/idx/sdk/api/client/WrapperUtil.java (hunk 5 above)
Comment (arvindkrishnakumar-okta) [reply api/src/main/java/com/okta/idx/sdk/api/client/WrapperUtil.java:36]: This should stay. This is the actual error from exception trace and would be useful to the developer.

Comment (arvindkrishnakumar-okta) [reply api/src/main/java/com/okta/idx/sdk/api/client/WrapperUtil.java]: removed

**Pull request merged**